tests/test__collection_api.py

```diff
@@ -1645,6 +1645,27 @@ def test_sparse_unique_index_dup(self):
 
         self.assertEqual(self.db.collection.find({}).count(), 1)
 
+    def test_unique_index_partial_filter_expression(self):
```

Comment:
Great test, can you try to add the same in test__mongomock to ensure we are consistent with `pymongo + MongoDB` please?

Reply:
Sorry I wasn't clear enough: please add the equivalent test, in a `_CollectionComparisonTest` test suite. Calls should use `self.cmp.do` and the expectations should not the be on the actual result but that the results match between mongomock and pymongo. I hope that it's clearer.

Reply:
This is still unresolved